test/integration/dialects/postgres/query-interface.test.js

```diff
@@ -6,15 +6,145 @@ var chai = require('chai')
   , Support = require(__dirname + '/../../support')
   , dialect = Support.getTestDialect()
   , DataTypes = require(__dirname + '/../../../../lib/data-types')
+  , Sequelize = require('../../../../index')
+  , Promise = Sequelize.Promise
   , _ = require('lodash');
 
+
 if (dialect.match(/^postgres/)) {
   describe('[POSTGRES Specific] QueryInterface', function () {
     beforeEach(function () {
       this.sequelize.options.quoteIdenifiers = true;
       this.queryInterface = this.sequelize.getQueryInterface();
     });
 
+    describe('createFunction', function () {
+
+      beforeEach( function () {
+        //make sure we don't have a pre-existing function called create_job
+        //this is needed to cover the edge case of afterEach not getting called because of an unexpected issue or stopage with the 
+        //test suite causing a failure of afterEach's cleanup to be called.
+        return this.queryInterface.dropFunction('create_job',[{type:'varchar',name:'test'}])
+        //suppress errors here. if create_job doesn't exist thats ok.
+        .catch( err => {});
+      });
+
+      afterEach( function () {
```

Comment:
Ok then we should add `after` rather than `afterEach`. With this the test will cleanup after each suite and same code will not run in each case.